app/functional_tests_fixtures/__init__.py

```diff
@@ -0,0 +1,503 @@
+    INBOUND_SMS_TYPE,
+    LETTER_TYPE,
+    SECOND_CLASS,
+    SEND_EMAILS,
+    SEND_LETTERS,
+    SEND_TEXTS,
+    VIEW_ACTIVITY,
+)
+from app.dao.annual_billing_dao import (
+    set_default_free_allowance_for_service,
+)
+from app.dao.api_key_dao import (
+    get_model_api_keys,
+    save_model_api_key,
+)
+from app.dao.inbound_numbers_dao import (
+    dao_get_inbound_number_for_service,
+    dao_set_inbound_number_to_service,
+)
+from app.dao.organisation_dao import (
+    dao_create_organisation,
+    dao_get_organisation_by_id,
+    dao_get_organisations_by_partial_name,
+    dao_update_organisation,
+)
+from app.dao.permissions_dao import permission_dao
+from app.dao.service_email_reply_to_dao import (
+    add_reply_to_email_address_for_service,
+    dao_get_reply_to_by_service_id,
+)
+from app.dao.service_inbound_api_dao import get_service_inbound_api_for_service, save_service_inbound_api
+from app.dao.service_permissions_dao import (
+    dao_add_service_permission,
+    dao_fetch_service_permissions,
+)
+from app.dao.service_sms_sender_dao import (
+    dao_add_sms_sender_for_service,
+    dao_get_sms_senders_by_service_id,
+)
+from app.dao.services_dao import (
+    dao_add_user_to_service,
+    dao_create_service,
+    dao_fetch_service_by_id,
+    get_services_by_partial_name,
+)
+from app.dao.templates_dao import dao_create_template, dao_get_all_templates_for_service
+from app.dao.users_dao import get_user_by_email, save_model_user
+from app.models import (
+    InboundNumber,
+    Organisation,
+    Permission,
+    Service,
+    ServiceEmailReplyTo,
+    ServiceInboundApi,
+    User,
+)
+from app.schemas import api_key_schema, template_schema
+
+functional_tests_fixtures_blueprint = Blueprint("functional_test_fixtures", __name__)
+
+"""
+This flask task is used to create a set of db fixtures that the functional tests use to run
+against.
+
+It is only intended to be used in non-production environments.
+
+The task will create all the database rows required for the funcitonal tests and output an
+environment file the functional tests can use to execute against the environment.
+
+The environment file can be outputed to a file or uploaded to AWS SSM. The file is intended for local
+testing and ssm for the pipeline.
+
+This flask task can be executed locally or via terraform for our AWS environments.
+
+It expects the following config to be set:
+    NOTIFY_ENVIRONMENT
+    MMG_INBOUND_SMS_USERNAME
+    MMG_INBOUND_SMS_AUTH
+    SQLALCHEMY_DATABASE_URI
+    REDIS_URL
+    SECRET_KEY
+    INTERNAL_CLIENT_API_KEYS
+    ADMIN_BASE_URL
+    API_HOST_NAME
+    FROM_NUMBER
+
+and the following environment variables:
+    REQUEST_BIN_API_TOKEN - request bin token to be used by functional tests
+    FUNCTIONAL_TEST_ENV_FILE - (optional) output file for the environment variables
+    SSM_UPLOAD_PATH - (optional) path to upload the environment variables to AWS SSM
+"""
+
+
+@functional_tests_fixtures_blueprint.cli.command("apply")
```

Comment:
one thing i've realised - we use the library `click` in `app/commands.py` to manage our commands - i think it make sense to move this decorated wrapper function there (we then dont need the blueprint at all so can remove the changes to `test_route_authentication`. (then we could invoke it via `flask command apply-functional-test-fixtures` or similar.

click will automatically pull in the docstring from a function to be used as the cli `--help` flag too so we could move the docstring to there - however im less sure about that as i think while running the command most people will probs have this file open in a code editor so the docstring being alongside the code seems more helpful to me intuitively

(i think the actual `apply_fixtures` func and all the create funcs should still live in this file)

Reply:
Good spot. Updated.